microceph/ceph/osd.go

```diff
@@ -438,28 +442,353 @@ func AddOSD(s *state.State, path string, wipe bool, encrypt bool) error {
 
 // ListOSD lists current OSD disks
 func ListOSD(s *state.State) (types.Disks, error) {
-	disks := types.Disks{}
+	return database.OSDQuery.List(s)
+}
+
+// RemoveOSD removes an OSD disk
+func RemoveOSD(s common.StateInterface, osd int64, bypassSafety bool, timeout int64) error {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*time.Duration(timeout))
+	defer cancel()
+	err := doRemoveOSD(ctx, s, osd, bypassSafety)
+	if err != nil {
+		// Checking if the error is a context deadline exceeded error
+		if errors.Is(err, context.DeadlineExceeded) {
+			return fmt.Errorf("timeout (%ds) reached while removing osd.%d, abort", timeout, osd)
+		}
+		return err
+	}
+	return nil
+
+}
+
+// sanityCheck checks if input is valid
+func sanityCheck(s common.StateInterface, osd int64) error {
+	// check osd is positive
+	if osd < 0 {
+		return fmt.Errorf("OSD must be a positive integer")
+	}
+
+	// check if the OSD exists
+	if !database.OSDQuery.HaveOSD(s.ClusterState(), osd) {
+		return fmt.Errorf("OSD %d not found", osd)
```

Comment:
We should probably consistently refer to OSDs throughout. In some cases its OSD $id, in other cases its osd.$id. I'd favor the osd.$id as its likely more familiar to users of Ceph and lines up Ceph docs and such.